apps/chef_objects/src/chef_key.erl

```diff
@@ -155,12 +164,28 @@ parse_binary_json(Bin, undefined) ->
     chef_object_base:validate_ejson(EJ, chef_object_base:public_key_spec(req)),
 
     % validate name field
-    chef_object_base:validate_ejson(EJ, {[ {<<"name">>, {string_match, chef_regex:regex_for(key_name)}},
-                                           {{req, <<"expiration_date">>}, string} ]}),
-
-    %% this will raise if expiration_date isn't a valid datetime
-    parse_expiration(ej:get({<<"expiration_date">>}, EJ)),
-    EJ.
+    chef_object_base:validate_ejson(EJ, {[
+                                           {<<"name">>, {string_match, chef_regex:regex_for(key_name)}}
+                                         ]}),
+
+    % validate that the expiration_date field is a ISO8601 UTC timestring (ending in Z) and that
+    % ec_date can parse it, and then turn it into a format sqerl's deps can handle
+    try
+        chef_object_base:validate_ejson(EJ, {[
```

Comment:
Since we want to standardize on this behavior, would it make sense to wrap it up into a `chef_object_base:validate_date_field(EJ, FieldName)` and `s/parse_expiration/parse_date`?  
(Or perhaps even into the sadly underutilized chef_json_validator)